bgpd/bgp_evpn_vty.c

```diff
@@ -4522,9 +4522,9 @@ DEFUN (bgp_evpn_vni_rd,
 	if (!bgp)
 		return CMD_WARNING;
 
-	if (bgp->vrf_id != VRF_DEFAULT) {
+	if (!bgp->advertise_all_vni) {
```

Comment:
We are check this flag at quite a few places. Having a function in the lines of is_evpn_enabled() would be useful. Just a suggestion.

Reply:
The `is_evpn_enabled` function we already have only checks if evpn is enabled, but not if it is in the vrf the command was performed against. By checking the flag in the passed vrf, we hit two birds with one stone.

As for a separate function `vrf_is_evpn(bgp)`, it feels like overkill to check a flag but does conveys more meaning, and would ease the transition to another definition of "an evpn vrf".

Your call :)

Reply:
Done in f920dd6 for both bgpd and zebra.